game-core/src/main/java/games/strategy/engine/framework/startup/ui/panels/main/MainPanel.java

```diff
@@ -114,8 +114,8 @@ private void addChat() {
     remove(chatSplit);
     chatPanelHolder.removeAll();
     final ChatModel chat = chatPanelSupplier.get().orElse(null);
-    if (chat instanceof ChatPanel) {
-      chatPanelHolder.add((ChatPanel) chat, BorderLayout.CENTER);
+    if (chat instanceof Component) {
```

Comment:
This still checks for an implementation detail. The API of `ChatModel` does not have anything in common with `Component`, it's implementation that `ChatModel` also happens to be a `Component` (which is also bad as we are making our UI component literally also a model).